app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+
+  validates :name, presence: true
+  validates :posts_counter, comparison: { greater_than_or_equal_to: 0 }
+
```

Comment:
- [ ]  Kindly add validation to validate if post_counter is an integer or not. GOod job adding validations for it to be greater than 0. You can use
```ruby
validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```